bin_sort/BinSort.java

```diff
@@ -0,0 +1,75 @@
+import java.util.Random;
+
+class Node {
+	public double data;
+	public Node next;
+
+	public Node(double data) {
+		this.data = data;
+		next = null;
+	}
+}
+
+// Time Complexity O(n)-> avg case and O(n^2)-> Worst Case
+// constraints:- 0<=A[i]<1
+public class BinSort {
+	public static void binSort(Node[] b, double[] a) {
+		for (int i = 0; i < a.length; i++) {
+			int ins = (int) (a[i] * 10);

```

Comment:
@mohitkyadav ```Math.floor()``` return largest floating-point value that less than or equal to the argument....that is ```Math.floor(0.923)``` returns 0.0, but I need 9 actually, So its actually better to use above method.